### PR TITLE
Add live coding to ch1 exercises

### PR DESCRIPTION
**Pull request by joelostblom**

### CODE REVIEW

**Comment by joelostblom** [comment]
@rorywhite200 This is ready for review. It ended up being much more than just a small fix, but now we have live exercise with at least some type of solution checking, which I wanted in thebook before receiving the feedback from the students. You might have to build locally, I had some issues with the gh actions timeing out earlier today, not sure why. I will put together some questions to send to the students and ask for your review there too.

File: textbook/1_intro.qmd [hunk 1]
```diff
@@ -1114,31 +1115,174 @@ Create a scatter plot of the food supply and emissions data by following these s
...
+    ______
+    tooltip=______,
+)
+```
+
+:::: {.callout-default .ex-hint}
+
+
+::: {.callout-default .ex-solution}
+
```
Comment (rorywhite200) [inline textbook/1_intro.qmd:1263]:
```suggestion
:::: {.callout-default .ex-hint}
Take a closer look at the emissions data table - you'll need the correct column name for the tooltip.

::: {.callout-default .ex-solution}
```{python}
alt.Chart(food_emission).mark_point().encode(
    x='Total Food Supply (kg/capita/yr)',
    y='Emissions (Kg CO2 / kg product)',
    tooltip='Food Item',
)
```

```

**Review by rorywhite200 — COMMENTED** [review]
The live coding exercises are really cool! The first one is working well for me for Altair. I've made a few suggestions for the tooltip exercise. 

1. Updating the exercise name so it has access to the imports in setup
2. Adding a hint and solution for exercise

**Comment by joelostblom** [comment]
Great, thank you! I missed that. Committed your suggestions and merging

**Pull request merged**